refiner-lib/src/near_stream.rs

```diff
@@ -160,4 +167,48 @@ mod tests {
         );
         assert_eq!(tx_2.logs.len(), 1);
     }
+
+    #[test]
+    fn test_block_82654651_nonce() {
+        // load state snapshot and main objects
+        let db_dir = tempfile::tempdir().unwrap();
+        let engine_path = db_dir.path().join("engine");
+        let tracker_path = db_dir.path().join("tracker");
+        let chain_id = 1313161554_u64;
+
+        crate::storage::init_storage(engine_path.clone(), "aurora".into(), chain_id);
+        let mut ctx =
+            EngineContext::new(&engine_path, "aurora".parse().unwrap(), chain_id).unwrap();
+
+        let json_snapshot: JsonSnapshot = {
+            let json_snapshot_data = std::fs::read_to_string(
+                "blocks/sate_H7Bfh9qCzWbJW9acao8B2jFMTrkfc31toczmTcMv7hY7.json",
+            )
+            .unwrap();
+            serde_json::from_str(&json_snapshot_data).unwrap()
+        };
+        initialize_engine_state(&mut ctx.storage, json_snapshot).unwrap();
+
+        let tx_tracker = TxHashTracker::new(tracker_path, 0).unwrap();
+        let mut stream = NearStream::new(chain_id, None, ctx, tx_tracker);
+
+        // parameters of the test
+        let block: NEARBlock = {
+            let data = std::fs::read_to_string("blocks/block-82654651.json").unwrap();
+            serde_json::from_str(&data).unwrap()
+        };
+        let target_receipt_id = "H7Bfh9qCzWbJW9acao8B2jFMTrkfc31toczmTcMv7hY7";
+        let expected_nonce = 12773;
+
+        // run and assert
+        let aurora_block = stream.next_block(&block).pop().unwrap();
+
+        let target_aurora_tx = aurora_block
+            .transactions
+            .iter()
+            .find(|tx| tx.near_metadata.receipt_hash.to_string() == target_receipt_id)
```

Comment:
Wanted to ask you if you can think in a more efficient way of doing this. Maybe I should create `CryptoHash` object for the `target_receipt_id` and then use `==` without the `to_string()`?

Reply:
I was asking more like a general question, but giving that there is only one tx we can do that for sure.